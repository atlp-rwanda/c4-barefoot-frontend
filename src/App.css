@import url('https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap');

.navbar .logo {
  color: white;
  text-align: center;
  margin-top: 20px;
  margin-left: 20px;
  justify-content: space-around;
  justify-items: center;
}

.navbar .navlinks {
  margin-right: 0;
  text-align: right;
  float: right;
}
.navbar .navlinks ul {
  margin-top: 25px;
  display: flex;
  justify-content: space-around;
  /* margin-right: 10px; */
  float: right;
}
.navbar .navlinks ul li {
  list-style: none;
  justify-content: space-between;
  padding: 10px;
}
.navbar .navlinks ul li a {
  display: inline;
  color: white;
  text-decoration: none;
  opacity: 0.8;
  margin-right: 20px;
}
.home{
    align-self: center;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    justify-content: space-around;
    margin-top: 150px;
    width: 300px;
}
/* form {
  width: 50vw;
  border: 1px solid lightgray;
  padding-top: 30px;
  margin-top: 20px;
  justify-items: center;
  border-radius: 10px;
  justify-content: center;
  height: 19vw;
  float: center;
  margin-left: 23vw;
}
form input {
  width: 90%;
  height: 30px;
  margin-bottom: 10px;
  border: none;
  border-bottom: 2px solid lightgray;
}
form input:focus {
  border: none;
}
form button {
  border: none;
  cursor: pointer;
  background-color: rgb(28, 175, 224);
  width: 90%;
  padding: 8px;
  color: white;
  font-size: 1.5rem;
  border-radius: 5px;
}
form .login-btn {
  border: none;
  margin-top: 60px;
  cursor: pointer;
  background-color: rgb(28, 175, 224);
  width: 90%;
  padding: 8px;
  color: white;
  font-size: 1.5rem;
  border-radius: 5px;
}
 */
*{
  margin: 0;
   padding: 0;
}
