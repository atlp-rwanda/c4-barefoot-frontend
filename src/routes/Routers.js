import ResetPasswordEmailForm from '../components/resetPassword/ResetPasswordEmailForm';
import NewPassword from '../components/resetPassword/NewPassword'

import React from 'react';
import Login from '../components/views/Login'
import SignUp from '../components/signup';
import signup from '../components/views/Signup'
import verifyAccount from '../components/signup/verifyAccount'
import Landing from '../components/views/LandingPage'
import PageNotFound from '../components/views/PageNotFound'
import { Switch, Redirect } from 'react-router-dom';
import RouteWithLayout  from '../components/RouteWithLayout';
import Profile from '../components/views/Profile';
import {  DefaultLayout, AuthorizedUserLayout } from '../components/layouts';
import Logout from '../components/views/Logout';

const Routes = () => {
    return (
      <Switch>
        <Redirect
          exact
          from="/"
          to="/welcome"
        />
        <RouteWithLayout
          component={Landing}
          exact
          layout={DefaultLayout}
          path="/welcome"
        />
        <RouteWithLayout
          component={Login}
          exact
          layout={DefaultLayout}
          path="/login"
        />
        <RouteWithLayout
          component={Profile}
          exact
          layout={AuthorizedUserLayout}
          path="/profile"
        />
         <RouteWithLayout
          component={signup}
          exact
          layout={DefaultLayout}
          path="/signup"
        />
        <RouteWithLayout
          component={ResetPasswordEmailForm}
          exact
          layout={DefaultLayout}
          path="/forgetpassword"
        />
        <RouteWithLayout
          component={NewPassword}
          exact
          layout={DefaultLayout}
          path="/user/reset-password"
        />
        <RouteWithLayout
          component={Logout}
          exact
          layout={AuthorizedUserLayout}
          path="/logout"
        />
        <RouteWithLayout
          component={PageNotFound}
          exact
          layout={DefaultLayout}
          path="/PageNotFound"
        />
        <RouteWithLayout 
            path="/signup" 
            component={signup}
            layout={DefaultLayout}
        />
        <RouteWithLayout 
            path="/user/verification/:token" 
            component={ verifyAccount }
            layout={DefaultLayout}
        />
        <RouteWithLayout 
            path="/user/verification/" 
            component={ verifyAccount }
            layout={DefaultLayout}
        />
        
        <Redirect to="/PageNotFound" />
      </Switch>
    );
  };
  
  export default Routes;
