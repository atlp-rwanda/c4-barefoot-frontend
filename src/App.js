import React, { Component } from "react";
import Routers from './routes/Routers'
import { BrowserRouter as Router } from 'react-router-dom'

import history from "./history";

class App extends Component {
    render() {
        return (
            <Router history={history}>
                <Routers />
            </Router>
        )
    }
}
export default App;