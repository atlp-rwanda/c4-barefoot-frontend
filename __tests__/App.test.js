import React from 'react';
import App from '../src/App';
import history from '../src/history';
import {BrowserRouter as Router} from 'react-router-dom'
import Routers from '../src/routes/Routers';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<App />', () =>{

    it('matches snapshot', () => {
        const props = {
            history : jest.fn()
        }
        const wrapper = shallow(<App {...props} /> );
        let apptest = (<Router history={history}><Routers/></Router>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    
})