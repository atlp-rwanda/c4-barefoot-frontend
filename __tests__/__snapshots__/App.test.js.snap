// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<BrowserRouter
  history={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Routes />
</BrowserRouter>
`;
